### ISSUE TITLE
Provide a Metropolis independence proposal function

### ISSUE THREAD

**Issue by axch** [tensorflow/probability#218]
Idea for usage:
```
kernel = tfp.mcmc.RandomWalkMetropolis(
              target_log_prob_fn=likelihood.log_prob,
              new_state_fn=tfp.mcmc.independence_proposal(tfp.some.distribution),
              seed=124
             )
```

Though, it's not really a random walk at that point, because the proposal ignores the current state, so maybe that deserves a different FooMetropolis class name.

**Comment by davmre** [comment]
Do you have a reference describing what this is? Does 'resimulation' mean the same thing as an independence sampler (i.e., always propose from the given distribution ignoring the current state of the chain)?

**Comment by axch** [comment]
Yes.  At least, I meant constructing valid transition kernels from independence proposals.  I don't know how standard or non-standard a term that is.

**Comment by SiegeLordEx** [comment]
One somewhat hacky way of implementing this is by creating a new `next_state_fn` for `RandomWalkMetropolis` that ignores the current state. Usage would look like:

```python
kernel = tfp.mcmc.RandomWalkMetropolis(
  target_log_prob_fn=...,
  new_state_fn=resimulation_fn(tfd.Distribution))
```
